src/hooks/twilio.ts

```diff
@@ -167,18 +167,16 @@ export const useVideoRoomState = ({
   const [token, setToken] = useState<string>();
 
   useEffect(() => {
-    (async () => {
-      if (!userId || !!token || !roomName) return;
+    if (!userId || !!token || !roomName) return;
 
-      const response = await getVideoToken({
-        userId,
-        roomName,
-      });
-
-      if (!response) return;
+    getVideoToken({
+      userId,
+      roomName,
+    }).then((token) => {
+      if (!token) return;
 
-      setToken(response.data.token);
-    })();
+      setToken(token);
+    });
   }, [userId, roomName, token]);
```

Comment:
As called out in the previous thread about this, this still isn't handling the error state/showing that to the user in a nice way. The implementation of this error handling state would be dramatically simplified by using the `react-use` async hooks I mentioned in my original review comments.

Reply:
Add it after the implementation of https://github.com/sparkletown/internal-sparkle-issues/issues/630

Reply:
That's a suboptimal choice.. I already explained simple/minimal boilerplate ways of solving this, which I have now also documented again in https://github.com/sparkletown/internal-sparkle-issues/issues/630#issuecomment-833283315

Reply:
Failing silently for a user is terrible UX, and "we can't be bothered till we have some perfect future system" isn't an acceptable response to that.